src/aegis/src/aegis.erl

```diff
@@ -61,23 +60,25 @@ open(#{} = Db, Options) ->
     FDBKey = erlfdb_tuple:pack(?WRAPPED_KEY, DbPrefix),
     WrappedKey = erlfdb:wait(erlfdb:get(Tx, FDBKey)),
 
-    % Unwrap it
-    DbKey = aegis_keywrap:key_unwrap(?ROOT_KEY, WrappedKey),
+    Db1 = Db#{aegis => WrappedKey},
 
-    Db#{
-        aegis => DbKey
-    }.
+    case gen_server:call(?CACHE, {maybe_rewrap_key, Db1}) of
+        WrappedKey ->
+            Db1;
+        NewWrappedKey ->
+            FDBKey = erlfdb_tuple:pack(?WRAPPED_KEY, DbPrefix),
+            ok = erlfdb:set(Tx, FDBKey, NewWrappedKey),
+            Db1#{aegis => NewWrappedKey}
+    end.
 
 
 encrypt(#{} = _Db, _Key, <<>>) ->
     <<>>;
 
 encrypt(#{} = Db, Key, Value) when is_binary(Key), is_binary(Value) ->
-    #{
-        uuid := UUID,
-        aegis := DbKey
-    } = Db,
+    gen_server:call(?CACHE, {encrypt, Db, Key, Value}).
 
+encrypt(DbKey, UUID, Key, Value) ->
```

Comment:
do not like this. I'd rather move this code into aegis_key_cache, though I think aegis_server would be a better name, since it does more than cache keys.